packages/paste-core/components/timeline/src/TimelineItemGroup.tsx

```diff
@@ -0,0 +1,63 @@
+import { Box, type BoxProps } from "@twilio-paste/box";
+import { DetailText } from "@twilio-paste/detail-text";
+import * as React from "react";
+
+import { TimelineContext, TimelineGroupContext } from "./TimelineContext";
+import { TimelineItemIcon } from "./TimelineItemIcon";
+import type { TimelineItemProps } from "./types";
+
+export const TimelineItemGroup = React.forwardRef<
+  HTMLDivElement,
+  Omit<TimelineItemProps, "title"> & {
+    /**
+     * Overrides the default element name to apply unique styles with the Customization Provider.
+     *
+     * @default "TIMELINE_ITEM_GROUP"
+     * @type {BoxProps["element"]}
+     * @memberof TimelineProps
+     */
+    element?: BoxProps["element"];
+  }
+>(({ children, timestamp, icon, element = "TIMELINE_ITEM_GROUP" }, ref) => {
+  const { orientation } = React.useContext(TimelineContext);
+
+  if (!orientation) {
+    throw new Error("TimelineItemGroup must be used within a Timeline component");
+  }
+
+  return (
+    <Box
+      element={element}
+      ref={ref}
+      as="li"
+      position="relative"
+      display="flex"
+      columnGap="space50"
+      rowGap="space50"
+      flexShrink={0}
+      flexDirection={orientation === "horizontal" ? "column" : "row"}
+      paddingX="space0"
+    >
+      <Box
+        display="flex"
+        width={orientation === "horizontal" ? "initial" : "20px"}
+        flexDirection={orientation === "horizontal" ? "row" : "column"}
+        alignItems="center"
+      >
+        <TimelineItemIcon title="Current" icon={icon} />
+      </Box>
+
+      <Box>
+        <Box marginTop="space10" marginBottom="space30">
+          <DetailText marginTop="space0">{timestamp}</DetailText>
```

Comment:
question: @twilio-labs/design-systems-eng thoughts on adding a marginBottom prop to DetailText to clean up some of the extra spacing we have to do here?

I've seen 8px bottom spacing show up a few times with Detail Text compositions now (see https://paste.twilio.design/components/page-header#page-header-on-a-wizard-page) that I think it could make sense.

cc @roxanagomez

Reply:
my only consideration is that it's a little weird to have the default marginTop be not 0 and have to reset it and the default marginBottom be 0 and have to add it. otherwise into it

Reply:
true, ah well, maybe I'll care more or have a stronger opinion if it comes up again lol.